src/main.qml

```diff
@@ -56,60 +57,90 @@ Application {
             enabled: !timer.running
             height: parent.height
-            width: Dims.w(20)
-            model: 10
-            delegate: SpinnerDelegate { text: index }
+            width: parent.width * 0.16
+            model: 24
+            delegate: SpinnerDelegate { text: zeroPad(index) }
             onCurrentIndexChanged: if(enabled) seconds = secondLV.currentIndex + 60*minuteLV.currentIndex + 3600*hourLV.currentIndex
         }
 
-        Label {
-            text: ":"
-            anchors.verticalCenter: parent.verticalCenter
-            horizontalAlignment: Text.AlignHCenter
-            width: Dims.w(20)
-            font.pixelSize: Dims.l(12)
-        }
-
         Spinner {
             id: minuteLV
             currentIndex: 5
             enabled: !timer.running
             height: parent.height
-            width: Dims.w(20)
+            width: parent.width * 0.505
             model: 60
             highlightMoveDuration: currentIndex != 0 ? 400 : 0
             onCurrentIndexChanged: if(enabled) seconds = secondLV.currentIndex + 60*minuteLV.currentIndex + 3600*hourLV.currentIndex
         }
 
-        Label {
-            text: ":"
-            anchors.verticalCenter: parent.verticalCenter
-            horizontalAlignment: Text.AlignHCenter
-            width: Dims.w(20)
-            font.pixelSize: Dims.l(12)
-        }
-
         Spinner {
             id: secondLV
             currentIndex: 0
             enabled: !timer.running
             height: parent.height
-            width: Dims.w(20)
+            width: parent.width * 0.16
             model: 60
             highlightMoveDuration: currentIndex != 0 ? 400 : 0
             onCurrentIndexChanged: if(enabled) seconds = secondLV.currentIndex + 60*minuteLV.currentIndex + 3600*hourLV.currentIndex
         }
     }
 
+    Label {
+        text: "h"
+        anchors {
+            centerIn: app
+            verticalCenterOffset: -app.height * 0.056
+            horizontalCenterOffset: -app.width * 0.199
+        }
+        horizontalAlignment: Text.AlignHCenter
+        font.pixelSize: Dims.l(6.4)
+        font.styleName: "Medium"
+        opacity:0.9
+    }
+
+    Label {
+        text: "m"
+        anchors {
+            centerIn: app
+            verticalCenterOffset: -app.height * 0.056
+            horizontalCenterOffset: app.width * 0.084
+        }
+        horizontalAlignment: Text.AlignHCenter
+        font.pixelSize: Dims.l(6.4)
+        font.styleName: "Medium"
+        opacity:0.9
+    }
+
+    Label {
+        text: "s"
+        anchors {
+            centerIn: app
+            verticalCenterOffset: -app.height * 0.056
+            horizontalCenterOffset: app.width * 0.343
+        }
+        horizontalAlignment: Text.AlignHCenter
+        font.pixelSize: Dims.l(6.4)
+        font.styleName: "Medium"
+        opacity:0.9
+    }
+
     IconButton {
         id: iconButton
         iconName: timer.running ? "ios-pause" : "ios-timer-outline"
```

Comment:
Unrelated to this PR and not blocking it.

I wonder if the UX would be a bit clearer with a play icon instead of a ios-timer-outline.

Reply:
Good catch! The last remaining issue i had with the layout was the tiny spacing from the Spinner to the ios-icon.
I wanted to address that by making a second version of the Spinner with less height to be used in situations where the spinner needs to be visually shorter.
But changing the logo to something different might also work. Trying now.

Reply:
Oh wait, re spinner height, i see it is actually dynamic height. 
But only reduces the whole Spinner height using fade/mask effects.
Reducing height of the Spinners here made the touch area too small tho and it was harder to operate.
Looking for icon replacement now.

Reply:
Ok let’s keep the icon separate then.